src/m365/teams/commands/app/app-update.ts

```diff
@@ -27,30 +28,63 @@ class TeamsAppUpdateCommand extends GraphCommand {
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    const { id: appId, filePath } = args.options;
+    const { filePath } = args.options;
+
+    this
+      .getAppId(args)
+      .then((appId: string): Promise<any> => {
+        const fullPath: string = path.resolve(filePath);
+        if (this.verbose) {
+          logger.logToStderr(`Updating app with id '${appId}' and file '${fullPath}' in the app catalog...`);
+        }
+
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/appCatalogs/teamsApps/${appId}`,
+          headers: {
+            "content-type": "application/zip"
+          },
+          data: fs.readFileSync(fullPath)
+        };
+
+        return request
+          .put(requestOptions);
+      })
+      .then(_ => cb(), (res: any): void => this.handleRejectedODataJsonPromise(res, logger, cb));
+  }
 
-    const fullPath: string = path.resolve(filePath);
-    if (this.verbose) {
-      logger.logToStderr(`Updating app with id '${appId}' and file '${fullPath}' in the app catalog...`);
+  private getAppId(args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
     }
 
     const requestOptions: any = {
-      url: `${this.resource}/v1.0/appCatalogs/teamsApps/${appId}`,
+      url: `${this.resource}/v1.0/appCatalogs/teamsApps?$filter=displayName eq '${encodeURIComponent(args.options.name as string)}'`,
       headers: {
-        "content-type": "application/zip"
+        accept: 'application/json;odata.metadata=none'
       },
-      data: fs.readFileSync(fullPath)
+      responseType: 'json'
     };
 
-    request
-      .put(requestOptions)
-      .then(_ => cb(), (res: any): void => this.handleRejectedODataJsonPromise(res, logger, cb));
+    return request
+      .get<{ value: { id: string; }[] }>(requestOptions)
+      .then(response => {
+        const app: { id: string; } | undefined = response.value[0];
+
+        if (!app) {
+          return Promise.reject(`The specified Teams app does not exist`);
+        }
+
+        return Promise.resolve(app.id);
+      });
```

Comment:
What happens if you have two apps with the same name? I think we should return the error message that there are two and that you have to pick from one of these by id. In the output we should show the ids of the app